core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -66,6 +66,10 @@
     private static final boolean USE_OPENSSL = getBoolean("useOpenSsl", OpenSsl.isAvailable(),
                                                           value -> OpenSsl.isAvailable() || !value);
 
+    private static final int DEFAULT_NUM_MAX_CONNECTIONS = 10000; // from Tomcat default maxConnections
```

Comment:
I'm not sure about this default value, because we often get more connections than this. Could we keep using `Integer.MAX_VALUE`?

Reply:
BTW, what hardware baseline you used to match these default flag values?  Like, how many vCores with how much memory?

Reply:
Some of them are from Netty defaults. Some others are quite arbitrary although we tried to make it as sensible as possible. The connection timeout of 3.2 seconds is special though, which is related with the default value of some Linux socket parameter. Please feel free to let us know if you find any of them doesn't make sense.

Reply:
I'm thinking that maybe we could do some tests to make a baseline and recommendations tunning parameter values of Armeria. It will be very helpful for users of Armeria I think. 

Given hardware(or VM) env, with certain Armeria flag values configuration, provide some kind of performance benchmark report.

For example, for a GKE pod(4 GCP Core, 16GB Memory) with Armeria flag default values, and general web service load pattern(peek RPS, concurrency, cup load, etc. Maybe could use tools like `Load impact`), provide the performance test report to users.  

This approach will help us to decide default values for the flags based on solid evn and test feedbacks, also the performance test report could give the users a general clue of the server capacity. 

This is only an initial idea. Please feel free to share your ideas 😄

Reply:
I agree in general. Would be nice if we can handle this as a part of https://github.com/line/armeria/issues/799